packages/api/lambdas/sqs-message-consumer.js

```diff
@@ -61,23 +87,26 @@ async function processQueues(event, dispatchFn) {
       visibilityTimeout,
       deleteProcessedMessage: false,
     });
-    log.info(`processing queue ${queueUrl}`);
 
-    const messageConsumerFn = dispatchFn.bind({ rulesForQueue });
+    log.info(`Processing queue ${queueUrl}`);
+    const messageConsumerFn = dispatchFn.bind({ queueUrl, rulesForQueue });
+
     return consumer.consume(messageConsumerFn);
   }));
 }
 
 /**
- * Process an SQS message
+ * Archive and process an SQS message
  *
  * @param {string} queueUrl - Queue URL for incoming message
  * @param {Object} message - incoming queue message
  * @returns {Promise} - promise resolved when the message is dispatched
  */
-function dispatch(queueUrl, message) {
+async function dispatch(queueUrl, message) {
   const messageReceiveCount = Number.parseInt(message.Attributes.ApproximateReceiveCount, 10);
   const rulesForQueue = this.rulesForQueue;
+  log.info(`Archiving messages from queue ${queueUrl}`);
+  await archiveMessage(message);
```

Comment:
Can you refresh my memory: did we decide that we only want to implement this message archiving for SQS rules?

Reply:
We want to archive all messages but this ticket targets SQS messages. There is two other tickets for handling SNS and Kinesis ones. 
https://bugs.earthdata.nasa.gov/browse/CUMULUS-2370
https://bugs.earthdata.nasa.gov/browse/CUMULUS-2372